### PR TITLE
update hpke to not use rng with ecc when hardening is off

### PR DESCRIPTION
**Pull request by jpbland1**
# Description

update hpke to not use rng with ecc when hardening is off

# Testing

Commit test

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/src/hpke.c [hunk 1]
```diff
@@ -775,12 +776,17 @@ static int wc_HpkeEncap(Hpke* hpke, void* ephemeralKey, void* receiverKey,
...
+            rng = wc_rng_new(NULL, 0, hpke->heap);
+            wc_ecc_set_rng((ecc_key*)ephemeralKey, rng);
+#endif
 
             ret = wc_ecc_shared_secret((ecc_key*)ephemeralKey,
                 (ecc_key*)receiverKey, dh, &dh_len);
 
-            wc_rng_free(((ecc_key*)ephemeralKey)->rng);
+#ifdef ECC_TIMING_RESISTANT
+            wc_rng_free(rng);
```
Comment (dgarske) [inline wolfcrypt/src/hpke.c:788]: Set `rng = NULL`

File: wolfcrypt/src/hpke.c [hunk 2]
```diff
@@ -736,6 +736,7 @@ static int wc_HpkeEncap(Hpke* hpke, void* ephemeralKey, void* receiverKey,
     byte* sharedSecret)
 {
     int ret;
+    WC_RNG* rng;
```
Comment (dgarske) [inline wolfcrypt/src/hpke.c:739]: Wrap with `ifdef ECC_TIMING_RESISTANT`

File: wolfcrypt/src/hpke.c [hunk 3]
```diff
@@ -992,12 +999,17 @@ static int wc_HpkeDecap(Hpke* hpke, void* receiverKey, const byte* pubKey,
...
+                rng = wc_rng_new(NULL, 0, hpke->heap);
+                wc_ecc_set_rng((ecc_key*)receiverKey, rng);
+#endif
 
                 ret = wc_ecc_shared_secret((ecc_key*)receiverKey,
                     (ecc_key*)ephemeralKey, dh, &dh_len);
 
-                wc_rng_free(((ecc_key*)receiverKey)->rng);
+#ifdef ECC_TIMING_RESISTANT
+                wc_rng_free(rng);
```
Comment (dgarske) [inline wolfcrypt/src/hpke.c:1011]: Set `rng = NULL`.

File: wolfcrypt/src/hpke.c [hunk 4]
```diff
@@ -992,12 +999,17 @@ static int wc_HpkeDecap(Hpke* hpke, void* receiverKey, const byte* pubKey,
             case DHKEM_P256_HKDF_SHA256:
             case DHKEM_P384_HKDF_SHA384:
             case DHKEM_P521_HKDF_SHA512:
-                ((ecc_key*)receiverKey)->rng = wc_rng_new(NULL, 0, hpke->heap);
+#ifdef ECC_TIMING_RESISTANT
+                rng = wc_rng_new(NULL, 0, hpke->heap);
```
Comment (dgarske) [inline wolfcrypt/src/hpke.c:1003]: Why is there no NULL check here?

File: wolfcrypt/src/hpke.c [hunk 5]
```diff
@@ -775,12 +776,17 @@ static int wc_HpkeEncap(Hpke* hpke, void* ephemeralKey, void* receiverKey,
         case DHKEM_P256_HKDF_SHA256:
         case DHKEM_P384_HKDF_SHA384:
         case DHKEM_P521_HKDF_SHA512:
-            ((ecc_key*)ephemeralKey)->rng = wc_rng_new(NULL, 0, hpke->heap);
+#ifdef ECC_TIMING_RESISTANT
+            rng = wc_rng_new(NULL, 0, hpke->heap);
```
Comment (dgarske) [inline wolfcrypt/src/hpke.c:780]: Why is there no NULL check here?

File: wolfcrypt/src/hpke.c [hunk 6]
```diff
@@ -951,6 +957,7 @@ static int wc_HpkeDecap(Hpke* hpke, void* receiverKey, const byte* pubKey,
     word16 pubKeySz, byte* sharedSecret)
 {
     int ret;
+    WC_RNG* rng;
```
Comment (dgarske) [inline wolfcrypt/src/hpke.c:960]: Wrap with `ifdef ECC_TIMING_RESISTANT`

**Review by dgarske — CHANGES_REQUESTED** [review]
Did you test with `./configure --disable-harden`?

**Review by dgarske — APPROVED** [review]
Thanks, all good now.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**